File: practice/src/main/java/com/decade/practice/model/domain/entity/WelcomeEvent.java
package com.decade.practice.model.domain.entity;

import jakarta.persistence.DiscriminatorValue;
import jakarta.persistence.Entity;

@Entity
@DiscriminatorValue("HELLO WORLD")
public class WelcomeEvent extends TextEvent {

        protected WelcomeEvent() {
                super();
        }

        public WelcomeEvent(Chat chat, User admin, User user) {
                super(chat, admin, welcomeMessage(user));
                setOwner(user);
        }

        private static String welcomeMessage(User user) {
                return "HELLO " + user.getUsername();
        }
}

File: practice/src/main/java/com/decade/practice/model/entity/WelcomeEvent.kt
package com.decade.practice.model.entity

import jakarta.persistence.DiscriminatorValue
import jakarta.persistence.Entity

const val WELCOME = "HELLO WORLD"

private fun welcomeMessage(user: User): String {
      return "HELLO ${user.username}"
}

@Entity
@DiscriminatorValue(WELCOME)
class WelcomeEvent(
      chat: Chat,
      admin: User,
      user: User,
) : TextEvent(chat, admin, welcomeMessage(user)) {
      init {
            owner = user
      }
}
